.hooks/pre-commit

```diff
@@ -0,0 +1,6 @@
+#/bin/sh
+
+cargo clippy
+if [ $? -neq 0 ]; then
+    exit 1
+fi
```

Comment:
We also require the python files to be formatted using `flake` and `black`. Maybe worth adding them too?

Reply:
There are tons of errors when running `$(which flake8 || which flake) .` from the root